src/compiler/reflection/IntrospectStructPass.cpp

```diff
@@ -182,61 +171,61 @@ llvm::PreservedAnalyses IntrospectStructPass::run(llvm::Module& M, llvm::ModuleA
     }
   }
 
-  
-  llvm::SmallDenseMap<llvm::Function*, TypeInformation> StructTypeInfoForBuiltin;
-  llvm::SmallVector<llvm::CallInst*, 16> Calls;
-  for(auto* F: BuiltinInstantiations) {
-    for(auto* U : F->users()) {
-      if(llvm::CallInst* CI = llvm::dyn_cast<llvm::CallInst>(U)) {
+  llvm::SmallDenseMap<llvm::Function *, TypeInformation> StructTypeInfoForBuiltin;
+  llvm::SmallVector<llvm::CallInst *, 16> Calls;
+  for (auto *F : BuiltinInstantiations) {
+    for (auto *U : F->users()) {
+      if (llvm::CallInst *CI = llvm::dyn_cast<llvm::CallInst>(U)) {
         Calls.push_back(CI);
-        // Obtain type referenced by builtin calls. To workaround opaque pointers, this
-        // assumes that an alloca instruction with the type is present in the preceding code.
-        auto* AI = getPrevAllocaInst(CI);
-        if(AI) {
-          llvm::Type* T = AI->getAllocatedType();
-          auto TI = getTypeInformation(T, M);
-          StructTypeInfoForBuiltin[F] = TI;
+
+        if (auto *AI = llvm::dyn_cast<llvm::AllocaInst>(CI->getOperand(0))) {
+          llvm::Type *T = AI->getAllocatedType();
+          StructTypeInfoForBuiltin[F] = getTypeInformation(T, M);
+        } else if (auto *BI = llvm::dyn_cast<llvm::BitCastInst>(CI->getOperand(0))) {
+          llvm::Type *T = BI->getSrcTy()->getNonOpaquePointerElementType();
```

Comment:
The documentation on this says:
> Only use this method in code that is not reachable with opaque pointers, or part of deprecated methods that will be removed as part of the opaque pointers transition. 

So it might be a good idea to add an `isOpaquePointerTy()` check before invoking this API.
What would be even better is to have a recursive algorithm, which goes backwards through bitcasts (and perhaps getelementptr) until it hits an `AllocaInst`.